mmengine/optim/optimizer/default_constructor.py

```diff
@@ -344,23 +327,24 @@ def reduce_param_groups(params: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
     groups and merge duplicated groups. This approach speeds up multi-tensor
     optimizer significantly.
 
-    Ref: https://github.com/facebookresearch/detectron2/blob/main/detectron2/solver/build.py
+    References: https://github.com/facebookresearch/detectron2/blob/main/detectron2/solver/build.py
 
     Args:
         params (List[Dict[str, Any]]): The parameter groups.
 
     Returns:
         List[Dict[str, Any]]: The reorganized parameter groups.
     """  # noqa: E501
-    params = _expand_param_groups(params)
-    groups = defaultdict(
-        list)  # re-group all parameter groups by their hyperparams
+    groups: dict = dict()
+
+    def param_info_id(param_info):
+        return dumps(param_info)
+
     for item in params:
-        cur_params = tuple((x, y) for x, y in item.items() if x != 'params')
-        groups[cur_params].extend(item['params'])
-    ret = []
-    for param_keys, param_values in groups.items():
-        cur = {kv[0]: kv[1] for kv in param_keys}
-        cur['params'] = param_values
-        ret.append(cur)
-    return ret
+        param_info = {x: y for x, y in item.items() if x != 'params'}
+        info_id = param_info_id(param_info)
```

Comment:
Why do we not use a tuple as a key instead of dumping it?

Reply:
Of course, we extract the value of `lr`, `momentum`, and formulate them as a tuple, but will this format be easier to be understood?